src/common-ssh/sftp.c

```diff
@@ -653,10 +654,19 @@ static int guac_common_ssh_sftp_ls_ack_handler(guac_user* user,
 
         /* Determine mimetype */
         const char* mimetype;
-        if (LIBSSH2_SFTP_S_ISDIR(attributes.permissions))
-            mimetype = GUAC_USER_STREAM_INDEX_MIMETYPE;
-        else
-            mimetype = "application/octet-stream";
+        //adding file size and permission
+        char tmpstr[150];
+	if (LIBSSH2_SFTP_S_ISDIR(attributes.permissions)) {
+		sprintf(tmpstr, "{\"mime\":\"%s\",\"size\":%llu,\"perm\":%lu}",
```

Comment:
I'm not sure how JSON data is assembled elsewhere in the code - if `sprintf` is routinely used like this, or if there's a library that handles it? Might be worth checking that out - I'd be a little surprised if all of the other places in the guacd code that assemble JSON do it like this.

Reply:
We use the various internal `guac_common_json_*()` functions like `guac_common_json_write_property()`.